doc/src/contributing/new-contributors-guide/dev-setup.md

```diff
@@ -0,0 +1,228 @@
+to do them again.
+
+(dev-setup-create-github-account)=
+### Create a GitHub Account
+
+A [GitHub](https://github.com) account is required to contribute to SymPy. If
+you have not one yet then sign up at <https://github.com/signup/free>. Your
+GitHub account is your presence in the open source world, so we recommend
+choosing a professional username.
+
+### Setup SSH Keys
+
+To establish a secure connection between your computer and GitHub see detailed
+instructions in <https://help.github.com/en/articles/set-up-git> or at
+<https://help.github.com/en/articles/adding-a-new-ssh-key-to-your-github-account>.
+
+If you have any problems with SSH access to GitHub, read the troubleshooting
+instructions at <https://help.github.com/en/articles/troubleshooting-ssh>, or
+ask us on the [mailing list](https://groups.google.com/group/sympy).
+
+### Fork SymPy
+
+Create your own *fork* of the SymPy project on GitHub. If you have already
+done this before, you do not need to do it again.
+
+Go to the [SymPy GitHub repository](https://github.com/sympy/sympy) and click the **Fork** button.
+
+Now you have your own repository for the SymPy project. The address of the
+forked project will look something like
+`https://github.com/<your-github-username>/sympy`, where
+`<your-github-username>` is your GitHub username.
+
+## Get the SymPy Code
+
+It is recommended practice to create a fork of the SymPy project for your development purposes. Create your own fork of the SymPy project (if you have not yet). Go to the SymPy GitHub repository:
+
+```bash
+https://github.com/sympy/sympy
+```
+
+You will now have a fork at `https://github.com/<your-user-name>/sympy`.
+
+Then, on your machine browse to where you would like to store SymPy, and clone (download) the latest code from SymPy's original repository (about 77 MiB):
+
+```bash
+$ git clone https://github.com/sympy/sympy
+```
+
+Then assign your read-and-write repo to a remote called "github" (replace
+`<your-github-username>` with your GitHub username):
+
+```
+git remote add github git@github.com:<your-github-username>/sympy.git
+```
+
+For more information about GitHub forking and tuning see:
+<https://help.github.com/en/articles/about-pull-requests>, <https://help.github.com/en/articles/fork-a-repo>, and <https://help.github.com/en/articles/set-up-git>
+
+After the configuration, your setup should be similar to this:
+
+```bash
+$ git remote -v
+origin   https://github.com/sympy/sympy (fetch)
+origin   https://github.com/sympy/sympy (push)
+upstream https://github.com/github/sympy (fetch)
+upstream https://github.com/github/sympy (push)
+```
+
+## Virtual Environment Setup
+
+You may want to take advantage of using virtual environments to isolate your development version of SymPy from any system wide installed versions, e.g. from `apt-get install python-sympy`.
+
+We recommend using `conda` to create a virtual environment:
+
+```bash
+$ conda create -n sympy-dev python=3 mpmath flake8
+```
```

Comment:
I don't think that this should "recommend" conda since there is no real reason to use conda for sympy itself (although it can be useful for some optional dependencies).

A simple setup using pip and venv is like:
```
cd sympy
python -m venv .venv
source .venv/bin/activate
pip install -e .
```
Then when opening a new terminal for development only the `cd` and `source` steps are needed.

Reply:
Starting with conda is a good idea for other reasons, but I can change this to show both ways.

Reply:
Consider someone who already knows how to use virtual environments and pip (or whatever alternatives) and already knows what conda is but doesn't usually use it. There is no sympy-specific reason for that person to use conda so we should not give the impression that contributing to sympy is a reason for using conda.

Reply:
I see you've changed it already